### PR TITLE
Move filtering to content profile

### PR DESCRIPTION
**Pull request by ngfreiter**
Adds filtering and primary profile selection grammar to the content profile. Also adds the ability to specify no profile for an entire namespace, or to specify that all profiles in a given namespace are primary.
- Applying primary profile selection to namespace
    - `Namespace: obf *`
- Applying no profile selection to namespace
    - `Namespace: obf NP`
- Any mentioned entry/group is treated as being selected to make primary profile of, example:
    - `Observation:` 
    - Now observation will be a primary profile
- Applying a flag at the level of an individual entry overrides a namespace level flag
PR 1/5

### CODE REVIEW

File: src/main/antlr/SHRContentProfileParser.g4 [hunk 1]
```diff
@@ -8,10 +8,10 @@ doc:                docHeader contentsDefs;
 docHeader:          KW_GRAMMAR KW_G_CONTENT_PROFILE version;
 
 contentsDefs:       (namespaceHeader contentDefs)*;
-namespaceHeader:    KW_NAMESPACE namespace;
+namespaceHeader:    KW_NAMESPACE namespace (namespaceFlag)?;
```
Comment (cmoesel) [inline src/main/antlr/SHRContentProfileParser.g4]: The parens `(` `)` are not harmful, but seem unnecessary...  I'm guessing that maybe you originally had an `|` in there but it got removed?

**Review by cmoesel — COMMENTED** [review]
I'm assuming you've run this grammar by @markkramerus? The one thing I'm not sure about is:

> Any mentioned entry/group is treated as being selected to make primary profile of

This doesn't quite make sense because `MyProfile: NP` is a "mention" of MyProfile, but the mention is to suppress the profile -- so you certainly don't want it as primary.  Similarly, what if we add other capabilities in the future that might require us to mention a profile that we don't necessarily want to be primary?

I assume the above questions might need @markkramerus's input since I'm assuming he's the one who directed you to make it work this way.

**Comment by cmoesel** [comment]
Also, the current config has a `filterStrategy` and `primarySelectionStrategy` -- but this seems to address only primary selection?  Are they being combined somehow or is filtering coming later?

**Comment by ngfreiter** [comment]
The use of the word mention was too general there. To clarify, any mentioned profile that is not otherwise flagged as `NP` is treated as primary. If we add other flags in the future, we could define how those flags impact whether or not a profile is treated as primary, in the same way that we do for No Profile now. Alternatively we could make the flag for primary selection explicit. So that for each profile that we want to be primary, we have to say something like `MyProfile: PR`, assuming PR means primary. The current behavior is perhaps easier on the user, they don't have to retype the same flag over and over, but maybe also less clear and a bit less extensible than being explicit.

After talking about it @markkramerus and I settled on combining filtering and primary selection. So any profile mentioned that is not flagged as `NP` is not filtered out, and is selected as primary. I would think if you select a profile as primary, you would certainly not want to filter it out. So the only case would be if you for some reason wanted to not filter something out, but also not select it as primary. We didn't see a use for this, but could have missed something.

File: src/main/antlr/SHRContentProfileParser.g4 (hunk 1 above)
Comment (ngfreiter) [reply src/main/antlr/SHRContentProfileParser.g4]: Fixed in https://github.com/standardhealth/shr-grammar/pull/45/commits/56b015afeaf7aa9d489026ce54c8aa4011a79ecf.

**Comment by cmoesel** [comment]
Got it.  Thanks!  Looks good to me!

**Review by cmoesel — APPROVED** [review]

**Pull request merged**